ami/main/api/views.py

```diff
@@ -884,6 +901,15 @@ def get_queryset(self) -> QuerySet:
 
         return qs
 
+    # def retrieve(self, request: Request, *args, **kwargs) -> Response:
```

Comment:
This works and is very helpful in the UI. But consider if there is a cleaner way to do it. Basically we pre-calculate the occurrences count for list views, but bring back the occurrences_count method on the models for detail views, and omit the annotation in that case.